contribute/design/1.x/csi/20190606-csi-volume-provisioning.md

```diff
@@ -0,0 +1,251 @@
+---
+oep-number: draft CSI 20190606
+title: CSI Volume Provisioning
+authors:
+  - "@amitkumardas"
+owners:
+  - "@kmova"
+  - "@vishnuitta"
+editor: "@amitkumardas"
+creation-date: 2019-06-06
+last-updated: 2019-06-06
+status: provisional
+see-also:
+  - NA
+replaces:
+  - NA
+superseded-by:
+  - NA
+---
+
+# CSI Volume Provisioning
+
+## Table of Contents
+
+* [Table of Contents](#table-of-contents)
+* [Summary](#summary)
+* [Motivation](#motivation)
+    * [Goals](#goals)
+    * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+    * [User Stories](#user-stories)
+      * [Create a volume](#create-a-volume)
+      * [Delete the volume](#delete-the-volume)
+    * [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+    * [Risks and Mitigations](#risks-and-mitigations)
+* [Graduation Criteria](#graduation-criteria)
+* [Implementation History](#implementation-history)
+* [Drawbacks](#drawbacks)
+* [Alternatives](#alternatives)
+* [Infrastructure Needed](#infrastructure-needed)
+
+## Summary
+
+This proposal charts out the design details to implement Container Storage 
+Interface commonly referred to as **CSI** to create and delete openebs volumes.
+
+## Motivation
+
+CSI implementation provides the necessary abstractions for any storage provider
+to provision volumes and related use-cases on any container orchestrator. This
+is important from the perspective of all OpenEBS storage engines, since it will 
+make all OpenEBS storage engines adhere to CSI specifications and hence abstract
+the same from inner workings of a container orchestrator.
+
+### Goals
+
+- Ability to create and delete a volume in a kubernetes cluster using CSI
+- Enable higher order applications to consume this volume
+
+### Non-Goals
+
+- Providing tunables w.r.t creating and deleting a volume
+- Coupling CSI implementation with any particular container orchestrator e.g. Kubernetes
+- Ability to resize a volume
+- Ability to take volume snapshots
+- Ability to clone a volume
+- Handling high availability in cases of node restarts
+- Handling volume placement related requirements
+
+## Proposal
+
+### User Stories
+
+#### Create a volume
+As an application developer I should be able to provide a volume that can be 
+consumed by my application. This volume should get created dynamically during 
+application creation time.
+
+#### Delete the volume
+As an application developer I should be able to delete the volume that was being
+consumed by my application. This volume should get deleted only when the 
+application is deleted.
+
+### Implementation Details/Notes/Constraints
+
+There were previous attempts to implement CSI which is tightly coupled with the 
+way Kubernetes works. This attempt will revisit to have an implementation that
+is not tied to Kubernetes native resources.
```

Comment:
I can't relate this "is not tied to kubernetes native resources", as CSI itself is something that is tightly coupled with kubernetes (requires kubernetes APIs).

Reply:
CSI abstracts container orchestrators.
An Ideal CSI implementation should work with any container orchestrator e.g. mesos or k8s or nomad, etc

Reply:
in other words, how new approach will solve this ?